x-pack/test/functional/services/ml/data_frame_analytics_results.ts

```diff
@@ -61,6 +67,209 @@ export function MachineLearningDataFrameAnalyticsResultsProvider(
+        });
+      }
+    },
+
+    async toggleColumnSelectPopoverState(state: boolean) {
+      await retry.tryForTime(15 * 1000, async () => {
+        const popoverIsOpen = await testSubjects.exists(
+          '~mlExplorationDataGrid > ~dataGridColumnSelectorPopoverButton'
+        );
+        if (popoverIsOpen !== state) {
+          await testSubjects.click('~mlExplorationDataGrid > ~dataGridColumnSelectorButton');
+        }
+        await this.assertColumnSelectPopoverOpenState(state);
+      });
+    },
+
+    async hideAllResultsTableColumns() {
+      await this.toggleColumnSelectPopoverState(true);
+      await testSubjects.click('dataGridColumnSelectorHideAllButton');
+      await browser.pressKeys(browser.keys.ESCAPE);
+      await this.assertResultsTableEmpty();
+    },
+
+    async showAllResultsTableColumns() {
+      await this.toggleColumnSelectPopoverState(true);
+      await testSubjects.click('dataGridColumnSelectorShowAllButton');
+      await browser.pressKeys(browser.keys.ESCAPE);
+      await this.assertResultsTableNotEmpty();
+    },
+
+    async assertColumnSortPopoverOpenState(expectedOpenState: boolean) {
+      if (expectedOpenState === true) {
+        await testSubjects.existOrFail('~mlExplorationDataGrid > dataGridColumnSortingPopover', {
+          timeout: 30 * 1000,
+        });
+      } else {
+        await testSubjects.missingOrFail('~mlExplorationDataGrid > dataGridColumnSortingPopover', {
+          timeout: 30 * 1000,
+        });
+      }
+    },
+
+    async toggleColumnSortPopoverState(expectedState: boolean) {
+      await retry.tryForTime(15 * 1000, async () => {
+        const popoverIsOpen = await testSubjects.exists('dataGridColumnSortingSelectionButton');
+        if (popoverIsOpen !== expectedState) {
+          await testSubjects.click('~mlExplorationDataGrid > dataGridColumnSortingButton');
+        }
+        await this.assertColumnSortPopoverOpenState(expectedState);
+      });
+    },
+
+    async setColumnToSortBy(columnId: string, sortDirection: 'asc' | 'desc') {
+      await this.toggleColumnSortPopoverState(true);
+      await retry.tryForTime(15 * 1000, async () => {
+        // Pick fields to sort by
+        await testSubjects.existOrFail('dataGridColumnSortingSelectionButton');
+        await testSubjects.click('dataGridColumnSortingSelectionButton');
+        await testSubjects.existOrFail(`dataGridColumnSortingPopoverColumnSelection-${columnId}`);
+        await testSubjects.click(`dataGridColumnSortingPopoverColumnSelection-${columnId}`);
+        await testSubjects.existOrFail(`euiDataGridColumnSorting-sortColumn-${columnId}`);
+        // Click sorting direction
+        await testSubjects.click(
+          `euiDataGridColumnSorting-sortColumn-${columnId}-${sortDirection}`
+        );
+        // Close popover
+        await browser.pressKeys(browser.keys.ESCAPE);
+      });
+    },
+
+    async parseEuiDataGrid(tableSubj: string, maxColumnsToParse: number) {
```

Comment:
This and other methods added here seems to be largely the same as we already have in the `transform_table` and transform `wizard` services. Are we using the same data grid components under the hood in DFA and TFM? If so, I think it would make sense to just have the corresponding test methods once, e.g. in a new `ml.common_data_grid` service.

Reply:
Updated [here](https://github.com/elastic/kibana/pull/125162/commits/3f27f76acfb51aaa8629bcf5a6549844fc93b026)